utils/firestore.js

```diff
@@ -876,23 +880,24 @@ const filterDB = async (queries, siteCode, isParent) => {
     // Generate the queries.
     const generateQuery = (queryKeys) => {
         let participantQuery = collection;
-
-        for (let key in queryKeys) {
+        for (const key in queryKeys) {
             if (key === 'firstName' || key === 'lastName') {
                 const path = `query.${key}`;
                 const queryValue = key === 'firstName' ? queries.firstName : queries.lastName;
                 participantQuery = participantQuery.where(path, 'array-contains', queryValue);
             }
-            if (key === 'email' || key === 'phone') {
+            else if (key === 'email' || key === 'phone') {
                 const path = `query.${key === 'email' ? 'allEmails' : 'allPhoneNo'}`;
                 const queryValue = key === 'email' ? queries.email : queries.phone;
                 participantQuery = participantQuery.where(path, 'array-contains', queryValue);
             }
-            if (key === 'dob') participantQuery = participantQuery.where('371067537', '==', queries.dob);
-            if (key === 'connectId') participantQuery = participantQuery.where('Connect_ID', '==', parseInt(queries.connectId));
-            if (key === 'token') participantQuery = participantQuery.where('token', '==', queries.token);
-            if (key === 'studyId') participantQuery = participantQuery.where('state.studyId', '==', queries.studyId);
-            if (key === 'checkedIn') participantQuery = participantQuery.where('331584571.266600170.135591601', '==', 353358909);
+            else if (key === 'dob') participantQuery = participantQuery.where('371067537', '==', queries.dob);
+            else if (key === 'connectId') participantQuery = participantQuery.where('Connect_ID', '==', parseInt(queries.connectId));
+            else if (key === 'token') participantQuery = participantQuery.where('token', '==', queries.token);
```

Comment:
does dashboard code still allow for multiple filter keys to be sent in same call?

Reply:
It does. I noticed this is an optimization I missed when I rewrote this filterDB function 5 or so months ago. Only one key executes each iteration of the for loop, so the updated if/else if just short-circuits some checks.

I simulated testing with a single site code with isParent=false to emulate Biospecimen participant search
and used an array of site codes with isParent=true to simulate dashboard participant search. Local testing looked good, so I think it's on point.